### DISCUSSION TITLE
how to Counter intval according to condition

### DISCUSSION THREAD

**Discussion by jjqcat** [google/or-tools#4037] [CP-SAT questions] [answered]
Hi, I try to counter Intval according to some condition, as below
      std::vector<IntVar> X(10);
      for(i=0; i < 10; i++)
      {
          X[i] = model.NewIntVar(0, 10, 'X')                         # Decision Variable
       }
      A = model.NewIntVar(0, 100, 'A')                         # counter X[i] >1 

      How to count the numbers greater than 1 in array X？ such as condition  if( X[i] > 1) A++;          
    std::vector<IntVar> C(10);
    in other words, how to do cp_model.AddEquality(C[i], X[i]> cp_model.NewConstant(1))
    A = sum(C)

     cp_model.Maximize(A);

**Comment by jjqcat** [comment]
Or how to count the numbers less than 1 in array X? this code can't get valid result. cp_model.AddEquality(C[i], X[i] == cp_model.NewConstant(0));

**Comment by jjqcat** [comment]
The problem is:
cp_model.AddEquality(booleanA, x==y);
is that x==y "always" evaluates to false ,and also booleanA is false

**Comment by lperron** [comment, marked as answer]
https://github.com/google/or-tools/blob/stable/ortools/sat/docs/channeling.md

**Reply by jjqcat** [reply]
I solved this problem with your help ,thanks
